tests/test_vectorestore.py

```diff
@@ -328,3 +328,118 @@ def test_search_from_retriever_interface_with_filter() -> None:
     assert output == [Document(page_content="bar", metadata={"page": "1"})]
 
     docsearch.drop()
+
+
+def test_batch_insertion() -> None:
+    """Test batch insertion with different batch sizes."""
+    # Create a large number of documents
+    n = 100
+    texts = [f"text_{i}" for i in range(n)]
+    metadatas = [{"index": str(i)} for i in range(n)]
+    
+    # Test with default batch size
+    config = YDBSettings(drop_existing_table=True)
+    config.table = "test_ydb_batch_default"
+    docsearch1 = YDB.from_texts(
+        texts=texts,
+        embedding=ConsistentFakeEmbeddings(),
+        config=config,
+        metadatas=metadatas,
+    )
+    
+    # Test with small batch size
+    config = YDBSettings(drop_existing_table=True)
+    config.table = "test_ydb_batch_small"
+    docsearch2 = YDB.from_texts(
+        texts=texts,
+        embedding=ConsistentFakeEmbeddings(),
+        config=config,
+        metadatas=metadatas,
+        batch_size=10,  # Small batch size
+    )
+    
+    # Test with large batch size
+    config = YDBSettings(drop_existing_table=True)
+    config.table = "test_ydb_batch_large"
+    docsearch3 = YDB.from_texts(
+        texts=texts,
+        embedding=ConsistentFakeEmbeddings(),
+        config=config,
+        metadatas=metadatas,
+        batch_size=n,  # Entire dataset in one batch
+    )
+    
+    # Verify all documents are properly inserted with all batch sizes
+    for i, text in enumerate(texts):
+        # Test default batch size
+        results1 = docsearch1.similarity_search(
+            text, k=1, filter={"index": str(i)}
+        )
+        assert len(results1) == 1
+        assert results1[0].page_content == text
+        assert results1[0].metadata["index"] == str(i)
+        
+        # Test small batch size
+        results2 = docsearch2.similarity_search(
+            text, k=1, filter={"index": str(i)}
+        )
+        assert len(results2) == 1
+        assert results2[0].page_content == text
+        assert results2[0].metadata["index"] == str(i)
+        
+        # Test large batch size
+        results3 = docsearch3.similarity_search(
+            text, k=1, filter={"index": str(i)}
+        )
+        assert len(results3) == 1
+        assert results3[0].page_content == text
+        assert results3[0].metadata["index"] == str(i)
+    
+    # Clean up
+    docsearch1.drop()
+    docsearch2.drop()
+    docsearch3.drop()
+
+
+def test_batch_insertion_with_add_texts() -> None:
```

Comment:
We should test behavior of add_texts only. Other functions are just passing args to `add_texts`, so we should check only passing by mocks.